src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -187,9 +187,13 @@ public sealed partial class TokenCache : ITokenCacheInternal
                         requestParams.RequestContext.ApiEvent.DurationInCacheInMs += sw.ElapsedMilliseconds;
                     }
 
-                    // Don't cache PoP access tokens from broker
-                    if (msalAccessTokenCacheItem != null && !(response.TokenSource == TokenSource.Broker && response.TokenType == Constants.PoPAuthHeaderPrefix))
-                    {
+                    // Don't cache access tokens from broker
+#if iOS
```

Comment:
iOS broker gives MSAL back the RT, because iOS does not allow two apps to talk to each other in the background. So iOS broker is the only broker for which we have the flag `CanBrokerSupportSilentAuth` set to false. 

IMO, I'd prefer to have this logic at runtime rather than at build time, i.e. if (BrokerSupportsSilentAuth) { ... } but I understand that we may not want to create new broker just for this. Perhaps this flag is better to be set in PlatformProxy and not in Broker ? Thoughts @trwalke ?

Reply:
I support runtime check i.e. CanBrokerSupportSilentAuth rather than compile time check, if things change in the future regarding which OS supports the silent broker.

Reply:
this flag is already set in PlatformProxy so we are good